src/libraries/Common/tests/TestUtilities/System/AssertExtensions.cs

```diff
@@ -1081,46 +1121,62 @@ public static void Equal(Half expected, Half actual, Half variance)
             {
                 throw EqualException.ForMismatchedValues(ToStringPadded(expected), ToStringPadded(actual));
             }
+        }
+#endif
 
-            static unsafe bool IsNegativeZero(Half value)
+        /// <summary>Verifies that two <see cref="double"/> values's binary representations are identical.</summary>
+        /// <param name="expected">The expected value</param>
+        /// <param name="actual">The value to be compared against</param>
+        /// <exception cref="EqualException">Thrown when the representations are not identical</exception>
+        public static void Equal(double expected, double actual)
+        {
+            if (BitConverter.DoubleToInt64Bits(expected) == BitConverter.DoubleToInt64Bits(actual))
             {
-                return (*(ushort*)(&value)) == 0x8000;
+                return;
             }
 
-            static unsafe bool IsPositiveZero(Half value)
+            throw EqualException.ForMismatchedValues(ToStringPadded(expected), ToStringPadded(actual));
+        }
+
+        /// <summary>Verifies that two <see cref="float"/> values's binary representations are identical.</summary>
+        /// <param name="expected">The expected value</param>
+        /// <param name="actual">The value to be compared against</param>
+        /// <exception cref="EqualException">Thrown when the representations are not identical</exception>
+        public static void Equal(float expected, float actual)
+        {
+            if (BitConverter.SingleToInt32Bits(expected) == BitConverter.SingleToInt32Bits(actual))
```

Comment:
This is failing for the .NET Framework leg since `SingleToInt32Bits` is .NET Core exclusive. You'll need a shim to handle this.

Reply:
I guess, it's done, could you review?